### PR TITLE
Fix usage of deprecated func, fix logic in merging exponential histogram

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: processor/metricstransformprocessor/operation_aggregate_labels.go [hunk 1]
```diff
@@ -313,13 +317,15 @@ func mergeExponentialHistogramDataPoints(dpsMap map[string]pmetric.ExponentialHi
 			if dp.HasMax() && dp.Max() < dps.At(i).Max() {
 				dp.SetMax(dps.At(i).Max())
 			}
-			for b, bc := range dps.At(i).Negative().MBucketCounts() {
-				dps.At(i).Negative().MBucketCounts()[b] = dps.At(i).Negative().MBucketCounts()[b] + bc
+			for b := 0; b < dps.At(i).Negative().BucketCounts().Len(); b++ {
+				negatives[b] += dps.At(i).Negative().BucketCounts().At(b)
```
Comment (dmitryax) [inline processor/metricstransformprocessor/operation_aggregate_labels.go:321]: `negatives` is a copy of `dp.Negative().BucketCounts().AsRaw()`. so `dp.Negative().BucketCounts().AsRaw()` should no be used here

Comment (bogdandrutu) [reply processor/metricstransformprocessor/operation_aggregate_labels.go:321]: Why do you double them?

Comment (bogdandrutu) [reply processor/metricstransformprocessor/operation_aggregate_labels.go:321]: Shouldn't the logic be sum of all counts in all "dps"?

Comment (dmitryax) [reply processor/metricstransformprocessor/operation_aggregate_labels.go:321]: Oh nvm. I misread this, sorry

**Review by dmitryax — APPROVED** [review]

File: unreleased/metricstransform-fix-merge-exponential-histograms.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: "metricstransformprocessor"
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Fix logic in merging exponential histogram.
+
+# One or more tracking issues related to the change
+issues: [12865]
```
Comment (dmitryax) [inline unreleased/metricstransform-fix-merge-exponential-histograms.yaml:11]: Is this a bugfix? I think it's noop for the end user. So I would skip changelog here

Comment (dmitryax) [reply unreleased/metricstransform-fix-merge-exponential-histograms.yaml:11]: Oh I see the bug now. Thanks for fixing it!

**Pull request merged**